all: don't use the deprecated io/ioutil package

tklauser opened the pull request:
With the release of Go 1.16, the `io/ioutil` has been deprecated (see
https://golang.org/doc/go1.16#ioutil). Thus, use the corresponding
symbols from the `io` and `os` packages.

Note: pushed two more commits to make sure the correct Go version is used for the BPF prog tests and in the GitHub actions.

gandro commented on pkg/crypto/certloader/certloader_test.go:
```diff
@@ -301,26 +300,22 @@ type tlsConfigFiles struct {
-	if err != nil {
-		t.Fatal("ioutil.TempDir", err)
-	}
+	dir = t.TempDir()
```
nice catch!

sayboras commented on bpf/Makefile:
```diff
@@ -267,7 +267,7 @@ tests/bpf_ct_tests.o: tests/bpf_ct_tests.ll
 .PHONY: go_prog_test
 go_prog_test: $(BPF_TEST)
-	sudo go test -tags privileged_tests ./tests/prog_test
+	sudo -E PATH="$(PATH)" $(GO) test -tags privileged_tests ./tests/prog_test
```
nit: this might cause some files (e.g. go/pkg/mod//etc) having root permission, maybe one alternative is like below

```
go test -tags privileged_tests -exec sudo -v ./tests/prog_test
```

tklauser replied:
Oh, nice :tada: TIL about `go test -exec`. Thanks @sayboras!

tklauser committed:
go.mod: bump Go version to 1.16

This should have been part of 0cf2cff55ba5 ("Update Go to 1.16").

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
bpf: use correct Go version for go_prog_test

Pass $PATH and use the $GO variable to make sure to use the Go version
intended by the user for the prog tests.

Also don't build the test binary as root, but invoke it as root. Thanks
to Tam Mach for pointing this out!

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
.github/workflows: install Go for datapath build tests

Parts of the test use Go, so make sure we use the same version as for
other tests (i.e. the one specified in GO_VERSION).

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tklauser committed:
all: don't use the deprecated io/ioutil package

With the release of Go 1.16, the io/ioutil has been deprecated (see
https://golang.org/doc/go1.16#ioutil). Thus, use the corresponding
symbols from the io and os packages.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

kkourt commented on pkg/crypto/certloader/fswatcher/fswatcher_test.go:
```diff
@@ -34,9 +33,7 @@ func Test(t *testing.T) {
-	tmp, err := ioutil.TempDir("", "")
-	c.Assert(err, IsNil)
-	defer os.RemoveAll(tmp)
+	tmp := c.MkDir()
```
`MkDir()` says:
>  Create a new temporary directory which is automatically removed after  the suite finishes running.

is the directory removed recursively as `os.RemoveAll()` does above? had a look at the code: https://github.com/go-check/check/blob/10cb98267c6cb43ea9cd6793f29ff4089c306974/check.go#L170 but could not figure out if the dir is recursively removed or not.

[... checks a bit more ...]

OK, seems that it is should work as desired:
https://github.com/go-check/check/blob/10cb98267c6cb43ea9cd6793f29ff4089c306974/check.go#L637

kkourt reviewed (approved):
LGTM, thanks!

christarazi reviewed (approved):
:rocket:
